src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,111 @@
+            }
+
+            var targetFramework = FrameworkDescription.Instance.TargetFramework;
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins with target framework '{0}'.", targetFramework);
+
+            string[] pluginFiles;
+
+            try
+            {
+                // TODO: Here additional metadata could be loaded (eg: for security and integrity)
+                // instead of just string path
+                pluginFiles = pluginsConfig.GetValue<JToken>($"['{targetFramework}']").ToObject<string[]>();
+            }
+            catch (ArgumentException ex)
+            {
+                Log.Warning(ex, "Could not parse list of plugin paths. Invalid plugin configuration provided.");
+
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            if (pluginFiles == null || !pluginFiles.Any())
+            {
+                Log.Information("Skipping plugins load. Could not find any plugins with target framework '{0}'.", targetFramework);
+
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            var loadedPlugins = TryLoadPlugins(pluginFiles);
+
+            Log.Information("Successfully loaded '{0}' plugin(s).", property: loadedPlugins.Count);
+
+            return loadedPlugins;
+        }
+
+        private static IReadOnlyCollection<IOTelExtension> TryLoadPlugins(string[] pluginFiles)
+        {
+            var loaded = new List<IOTelExtension>();
+
+            foreach (string file in pluginFiles)
+            {
+                string fullPath = Path.GetFullPath(file);
+
+                if (File.Exists(fullPath))
+                {
+                    try
+                    {
+                        Assembly pluginAssembly = Assembly.LoadFrom(fullPath);
+                        ICollection<IOTelExtension> extensions = GetExtensions(pluginAssembly);
+
+                        if (extensions != null && extensions.Any())
+                        {
+                            loaded.AddRange(extensions);
+
+                            Log.Information("Plugin assembly loaded '{0}'.", pluginAssembly.FullName);
+                        }
+                        else
+                        {
+                            Log.Warning("Could not load {0} from '{1}'.", nameof(IOTelExtension), pluginAssembly.FullName);
+                        }
+                    }
+                    catch (Exception ex) when (
+                        ExceptionUtil.IsAssemblyLoadException(ex) ||
+                        ExceptionUtil.IsDynamicInvocationException(ex))
+                    {
+                        Log.Warning(ex, "Plugin assembly could not be loaded. Skipping vendor plugin load.");
+                    }
+                }
+                else
+                {
+                    Log.Warning("Plugin path is defined but could not find the path '{0}'.", fullPath);
```

Comment:
As mentioned somewhere else for plugins I think the right call is to fail the start.